feed/feed-inbound/feed-inbound-1.0.0/atom_rss_feed_1.0.0/src/main/java/org/wso2/carbon/inbound/feedep/FeedRetrieval.java

```diff
@@ -0,0 +1,210 @@
+            filter.add(Constants.ENTRY);
+            filter.add(Constants.TITLE);
+            filter.add(Constants.ID);
+            filter.add(Constants.UPDATED);
+            filter.add(Constants.LINK);
+            filter.add(Constants.AUTHOR);
+        }
+        opts.setParseFilter(filter);
+    }
+
+    //check time interval
+    public void execute() {
+        long currentTime = (new Date()).getTime();
+        if (((lastRanTime + scanInterval) <= currentTime)) {
+            lastRanTime = currentTime;
+            if (log.isDebugEnabled()) {
+                log.debug("lastRanTime " + lastRanTime);
+            }
+            consume();
+        } else if (log.isDebugEnabled()) {
+            log.debug("Skip cycle since concurrent rate is higher than the scan interval : Feed Inbound EP " + pathName);
+        }
+        if (log.isDebugEnabled()) {
+            log.debug("End : Feed Inbound EP : " + pathName);
+        }
+    }
+
+    //consume feeds
+    private void consume() {
+        InputStream input = null;
+        try {
+            input = new URL(host).openStream();
+            Document<Feed> doc;
+            doc = parser.parse(input, "", opts);
+            if (doc.getRoot() == null) {
+                log.error("Please check host address or feed type");
+                throw new SynapseException("Please check host address or feed type");
+            }
+            //convert RSS feeds as Atom
+            if (feedType.equalsIgnoreCase(FeedEPConstant.FEED_TYPE_RSS)) {
+                Factory factory = Abdera.getNewFactory();
+                feed = factory.newFeed();
+                OMElement item = (OMElement) doc.getRoot();
+                Iterator itemValue = item.getFirstElement().getChildrenWithName(FeedEPConstant.FEED_ITEM);
+                while (itemValue.hasNext()) {
+                    Entry entry = feed.insertEntry();
+                    OMElement omElement = (OMElement) itemValue.next();
+
+                    Iterator titleValue = omElement.getChildrenWithName(FeedEPConstant.FEED_TITLE);
+                    OMElement Title = (OMElement) titleValue.next();
+                    entry.setTitle(Title.getText());
+
+                    Iterator dateValue = omElement.getChildrenWithName(FeedEPConstant.FEED_PUBDATE);
+                    OMElement Updated = (OMElement) dateValue.next();
+                    Date date;
+                    try {
+                        date = format.parse(Updated.getText());
+                        entry.setUpdated(date);
+                    } catch (ParseException e) {
+                        if (!StringUtils.isEmpty(dateFormat)) {
+                            format = new SimpleDateFormat(dateFormat, Locale.ENGLISH);
+                            try {
+                                date = format.parse(Updated.getText());
+                            } catch (ParseException e1) {
+                                log.error("Error while parse date", e);
```

Comment:
You catch the exception as e1 and print and throw another ParseException (caught in above step). And use proper name for e1.